src/main/java/com/hms_networks/americas/sc/extensions/system/threading/SCCountdownLatch.java

```diff
@@ -0,0 +1,43 @@
+package com.hms_networks.americas.sc.extensions.system.threading;
+
+/**
+ * Class implementation of a CountDownLatch for Java 1.4.
+ *
+ * @since 1.4.0
+ * @author HMS Networks, MU Americas Solution Center
+ */
+public class SCCountdownLatch {
+
+  /** The count of the latch. */
+  private int count;
+
+  /**
+   * Constructs a new {@code SCCountdownLatch} initialized with the given count.
+   *
+   * @param count the number of times {@link #countDown} must be invoked before threads can pass
+   *     through {@link #await}.
+   * @throws IllegalArgumentException if {@code count} is negative.
```

Comment:
Does this throw IllegalArgumentException? I don't see where this would happen. I think some sort of check is required.

Reply:
Duplicate of Tom K's comment above - will address

Reply:
fixed